src/coreclr/jit/copyprop.cpp

```diff
@@ -288,6 +273,71 @@ unsigned Compiler::optIsSsaLocal(GenTreeLclVarCommon* lclNode)
     return lclNum;
 }
 
+//------------------------------------------------------------------------------
+// optCopyPropPushDef: Push the new live SSA def on the stack for "lclNode".
+//
+// Arguments:
+//    asg        - The assignment node for this def (will be "nullptr" for "use" defs)
+//    lclNode    - The local tree representing "the def" (that can actually be a use)
+//    lclNum     - The local's number (see "optIsSsaLocal")
+//    curSsaName - The map of local numbers to stacks of their defs
+//
+void Compiler::optCopyPropPushDef(GenTreeOp*           asg,
+                                  GenTreeLclVarCommon* lclNode,
+                                  unsigned             lclNum,
+                                  LclNumToLiveDefsMap* curSsaName)
+{
+    assert((lclNum != BAD_VAR_NUM) && (lclNum == optIsSsaLocal(lclNode)));
+
+    // Shadowed parameters are special: they will (at most) have one use, that is one on the RHS of an
+    // assignment to their shadow, and we must not substitute them anywhere. So we'll not push any defs.
+    if ((gsShadowVarInfo != nullptr) && lvaGetDesc(lclNum)->lvIsParam &&
+        (gsShadowVarInfo[lclNum].shadowCopy != BAD_VAR_NUM))
+    {
+        return;
```

Comment:
Should we assert here that `curSsaName` has no entry for `lclNum`?

Reply:
Yes. We must either push none of the defs for a given local, or all of them.